Applications that were created with earlier HDF5 releases may not compile
with 1.12 by default. The API Compatibility Macros in HDF5 allow users to
work around this issue. See: 
https://portal.hdfgroup.org/documentation/hdf5-docs/release_specifics/Migrating_from_HDF5_1.10_to_HDF5_1.12.html

--- ncl-6.6.2.orig/ni/src/ncl/yMakefile	2019-02-28 00:44:39.000000000 +0100
+++ ncl-6.6.2/ni/src/ncl/yMakefile	2024-07-14 08:46:43.786136149 +0200
@@ -100,7 +100,7 @@
 #endif
 
 #ifdef BuildHDF5
-HDF5_DEFS = -DBuildHDF5
+HDF5_DEFS = -DBuildHDF5 -DH5_USE_110_API
 HDF5_SRCS = NclNewHDF5.c NclHDF5.c h5reader.c h5writer.c
 HDF5_OBJS = NclNewHDF5.o NclHDF5.o h5reader.o h5writer.o
 #else
--- ncl-6.6.2.orig/ni/src/lib/nio/yMakefile	2019-02-28 00:44:39.000000000 +0100
+++ ncl-6.6.2/ni/src/lib/nio/yMakefile	2024-07-14 08:47:28.551405834 +0200
@@ -66,7 +66,7 @@
 #endif
 
 #ifdef BuildHDF5
-HDF5_DEFS = -DBuildHDF5
+HDF5_DEFS = -DBuildHDF5 -DH5_USE_110_API
 HDF5_SRCS = NclHDF5.c NclNewHDF5.c h5reader.c h5writer.c
 HDF5_OBJS = NclHDF5.o NclNewHDF5.o h5reader.o h5writer.o
 #else
